src/udev/udev-builtin-net_id.c

```diff
@@ -600,6 +603,93 @@ static int names_platform(sd_device *dev, NetNames *names, bool test) {
         return 0;
 }
 
+#define DEVICETREE_ALIASES "/proc/device-tree/aliases"
+
+static int dev_devicetree_onboard(sd_device *dev, NetNames *names) {
+        const char *alias, *ofnode_path, *ofnode_syspath, *parent_syspath;
+        _cleanup_free_ char *devicetree_syspath = NULL;
+        sd_device *aliases, *ofnode, *parent;
+        int r;
+        unsigned index;
+
+        if (!naming_scheme_has(NAMING_DEVICETREE_ALIASES))
+                return 0;
+
+        /* check if our direct parent has an of_node */
+        r = sd_device_get_parent(dev, &parent);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_syspath(parent, &parent_syspath);
+        if (r < 0)
+                return r;
+
+        r = sd_device_new_child(&ofnode, parent, "/of_node");
```

Comment:
ofnode is leaked. Must be defined with `_cleanup_(sd_device_unrefp)`

Reply:
> ofnode is leaked. Must be defined with _cleanup_(sd_device_unrefp)

What are the rules regarding this? `sd_device_get_parent` does not require this. Is it the `_new` vs `_get`?

> Please drop "/".

OK

Reply:
> > ofnode is leaked. Must be defined with _cleanup_(sd_device_unrefp)
> 
> What are the rules regarding this? `sd_device_get_parent` does not require this. Is it the `_new` vs `_get`?

Right. The parent device given by `sd_device_get_parent()` is owned by the child device, and will be freed when the child is freed. So, the caller must not free the parent device.
The child device created by `sd_device_new_child()` is owned by the caller, hence the caller must free it.